bindings/java/src/org/sleuthkit/datamodel/BlackboardArtifact.java

```diff
@@ -169,19 +170,63 @@ public String getDisplayName() {
 	 * @throws TskCoreException if there is a problem creating the description.
 	 */
 	public String getShortDescription() throws TskCoreException {
-		BlackboardAttribute attr = null;
-		if (artifactTypeId == ARTIFACT_TYPE.TSK_WEB_BOOKMARK.getTypeID()) {
-			attr = getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_URL));
+		List<BlackboardAttribute> attrs;
+		attrs = new ArrayList<BlackboardAttribute>();
+		if (artifactTypeId == ARTIFACT_TYPE.TSK_WEB_BOOKMARK.getTypeID()
+				|| artifactTypeId == ARTIFACT_TYPE.TSK_WEB_COOKIE.getTypeID()
+				|| artifactTypeId == ARTIFACT_TYPE.TSK_WEB_DOWNLOAD.getTypeID()
+				|| artifactTypeId == ARTIFACT_TYPE.TSK_WEB_HISTORY.getTypeID()) {
+			attrs.add(getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DOMAIN)));
 		} else if (artifactTypeId == ARTIFACT_TYPE.TSK_KEYWORD_HIT.getTypeID()) {
-			attr = getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_KEYWORD_PREVIEW));
-		} else if (artifactTypeId == ARTIFACT_TYPE.TSK_CALLLOG.getTypeID()) {
-			attr = getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_FROM));
+			attrs.add(getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_KEYWORD_PREVIEW)));
+		} else if (artifactTypeId == ARTIFACT_TYPE.TSK_DEVICE_ATTACHED.getTypeID()) {
+			attrs.add(getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DEVICE_ID)));
+		} else if (artifactTypeId == ARTIFACT_TYPE.TSK_CONTACT.getTypeID()
+				|| artifactTypeId == ARTIFACT_TYPE.TSK_MESSAGE.getTypeID()
+				|| artifactTypeId == ARTIFACT_TYPE.TSK_CALLLOG.getTypeID()) {
+			BlackboardAttribute name = getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_NAME));
+			if (name != null && !name.getDisplayString().isEmpty()) {
+				attrs.add(name);
+			}
+			if (null != getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER))) {
```

Comment:
For each of the following, rather than calling getAttribute() twice would it be cleaner to do something like:
if  (attr = getAttribute(new BlackboardAttribute.Type(TSK_PHONE_NUMBER) != null) {
    attrs.add(attr);
}

I'd also recommend importing the attribute types individually to make it easier to read, i.e. TSK_PHONE_NUMBER instead of BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER.